rust/noosphere-core/src/tracing.rs

```diff
@@ -1,27 +1,390 @@
+                (None, NoosphereLogFormat::Verbose)
+            } else {
+                (Some(NoosphereLogLevel::Info), NoosphereLogFormat::Minimal)
+            };
+
+        if let Some(noosphere_log) = noosphere_log {
+            noosphere_log_level_default = Some(noosphere_log.clone().into());
+            noosphere_log_format_default = noosphere_log.into();
+        }
+
+        let noosphere_log_level = match noosphere_log_level_env {
+            Some(noosphere_log_level_env) => match noosphere_log_level_env.parse() {
+                Ok(noosphere_log_level) => Some(noosphere_log_level),
+                _ => noosphere_log_level_default,
+            },
+            None => noosphere_log_level_default,
+        };
+
+        let noosphere_log_format = match noosphere_log_format_env {
+            Some(noosphere_log_format_env) => match noosphere_log_format_env.parse() {
+                Ok(noosphere_log_format) => noosphere_log_format,
+                _ => noosphere_log_format_default,
+            },
+            None => noosphere_log_format_default,
+        };
+
+        let mut env_filter = EnvFilter::default();
+        let mut rust_log_directives = rust_log_env
+            .unwrap_or_default()
+            .split(',')
+            .filter(|s| !s.is_empty())
+            .map(|s| s.parse())
+            .collect::<Result<Vec<Directive>, _>>()
+            .unwrap_or_else(|_| Vec::new());
+
+        let mut directives: Vec<Directive> = match noosphere_log_level {
+            Some(noosphere_log_level) => noosphere_log_level.into(),
+            None => Vec::new(),
+        };
+
+        directives.append(&mut rust_log_directives);
+
+        for directive in directives {
+            env_filter = env_filter.add_directive(directive)
+        }
+
+        let subscriber = tracing_subscriber::registry().with(env_filter);
+
+        match noosphere_log_format {
+            NoosphereLogFormat::Minimal => subscriber
+                .with(
+                    Layer::default().event_format(NoosphereMinimalFormatter::new(
+                        tracing_subscriber::fmt::format()
+                            .without_time()
+                            .with_target(false)
+                            .with_ansi(USE_ANSI_COLORS),
+                    )),
+                )
+                .init(),
+            NoosphereLogFormat::Verbose => subscriber
+                .with(tracing_subscriber::fmt::layer().with_ansi(USE_ANSI_COLORS))
+                .init(),
+            NoosphereLogFormat::Pretty => subscriber
+                .with(Layer::default().pretty().with_ansi(USE_ANSI_COLORS))
+                .init(),
+        };
+
+        Ok(())
+    }
+
+    /// A formatter designed to make `INFO` events display as closely as
+    /// possible to regular `println` output, while allowing arbitrary other
+    /// formatting for all other event types
+    struct NoosphereMinimalFormatter<F, S, N>(F, PhantomData<S>, PhantomData<N>)
```

Comment:
Very nice! Big blocker to switching over to smarter logs has been the verbosity of the default formatter